### PR TITLE
filter: add ratelimit log

### PR DESCRIPTION
**Pull request by zirain**
Commit Message: add ratelimt log
Additional Description: 
add some log to help debug ratelimt filter, in following cases specially:
- ratelimt Status Error due when timeout was set to small, refer to [istio PR](https://github.com/istio/istio/pull/33937)
- call grpc server onFailure

Risk Level: Low
Testing: manual testing
Docs Changes: 
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by wbpcode** [comment]
Please add signature for every commit by `git commit -s`.

File: source/extensions/filters/http/ratelimit/ratelimit.cc
Comment (wbpcode) [inline source/extensions/filters/http/ratelimit/ratelimit.cc:163]: For debugging logs, `debug` is more appropriate than `warn`.

File: source/extensions/filters/http/ratelimit/ratelimit.cc
Comment (wbpcode) [inline source/extensions/filters/http/ratelimit/ratelimit.cc:162]: `get` of `RequestHeaderMap` will not return `HeaderEntry` and there is no guarantee that the request id will be included in the request headers.

File: source/extensions/filters/http/ratelimit/ratelimit.cc
Comment (wbpcode) [inline source/extensions/filters/http/ratelimit/ratelimit.cc:163]: For debugging logs, `debug` is more appropriate than `warn`.
And we need to format this code by `tools/code_format/check_format.py fix`.

**Review by wbpcode — CHANGES_REQUESTED** [review]
Thanks. We do need some logs to help us to debug rate limit filter. 

I add some initial comments to help us to complete this.

File: source/extensions/filters/http/ratelimit/ratelimit.cc
Comment (zirain) [reply source/extensions/filters/http/ratelimit/ratelimit.cc:162]: so maybe we should remove request_id ?

Comment (wbpcode) [reply source/extensions/filters/http/ratelimit/ratelimit.cc]: May be you can use stream id as a alternative. Or you can use a placeholder if request id does not exist.

File: source/extensions/filters/common/ratelimit/ratelimit_impl.cc
Comment (davinci26) [inline source/extensions/filters/common/ratelimit/ratelimit_impl.cc:116]: I haven't seen this pattern anywhere else in the code base, from a quick code search I see that most cases in `source/extensions` use the `ENVOY_LOG` macro

**Review by davinci26 — CHANGES_REQUESTED** [review]
thanks! Got a small question. Please take a look at the format and DCO

File: source/extensions/filters/common/ratelimit/ratelimit_impl.cc
Comment (davinci26) [reply source/extensions/filters/common/ratelimit/ratelimit_impl.cc:116]: How would that log look like?

Comment (wbpcode) [reply source/extensions/filters/common/ratelimit/ratelimit_impl.cc:116]: `ENVOY_LOG_TO_LOGGER` is generally used in free functions or member methods of the class that does not inherit `Loggable`. It is no different from ordinary ENVOY_LOG in the format.

Comment (zirain) [reply source/extensions/filters/common/ratelimit/ratelimit_impl.cc:116]: hello, what's the conclusion？

Comment (davinci26) [reply source/extensions/filters/common/ratelimit/ratelimit_impl.cc:116]: Thanks for the explanation, shim

**Comment by zirain** [comment]
> /wait

do we still wait something else?

**Comment by davinci26** [comment]
Is the CI is passing the DCO and the format check?

**Comment by davinci26** [comment]
You will need to apply the DCO on the first commit (follow the instructions [here](https://github.com/envoyproxy/envoy/blob/main/CONTRIBUTING.md#dco-sign-your-work)) which would require rewritting the git history of your PR.

Also take a look at https://dev.azure.com/cncf/envoy/_build/results?buildId=87580&view=logs&j=55bc2ab9-36c1-5ab8-58a6-8c258728eeff&t=33b57bf9-ef13-58c8-bc2b-f804aeb16d39 which indicates that you have some format errors

**Comment by wbpcode** [comment]
> Please add signature for every commit by git commit -s.

> And we need to format this code by tools/code_format/check_format.py fix.

These are the remaining problems.

**Comment by zirain** [comment]
> > Please add signature for every commit by git commit -s.
> 
> > And we need to format this code by tools/code_format/check_format.py fix.
> 
> These are the remaining problems.

All checks have passed.

**Review by davinci26 — APPROVED** [review]
Thanks, LGTM @envoyproxy/senior-maintainers for a second review

**Review by wbpcode — DISMISSED** [review]
Thanks. And waiting senior-maintainers to merge it.

File: source/extensions/filters/common/ratelimit/ratelimit_impl.cc
Comment (mattklein123) [inline source/extensions/filters/common/ratelimit/ratelimit_impl.cc:117]: This will log spam at high volume if the upstream RL server is having issues which I don't think is what we want. I would switch this to debug. There should already be a failure stat. If not we can add one?

**Review by mattklein123 — CHANGES_REQUESTED** [review]
Thanks, small comment.

/wait

File: source/extensions/filters/common/ratelimit/ratelimit_impl.cc
Comment (zirain) [reply source/extensions/filters/common/ratelimit/ratelimit_impl.cc:117]: there's nothing record right now when  the upstream RL server is having issues, I think 'warn' is suitable in this scenario

Comment (mattklein123) [reply source/extensions/filters/common/ratelimit/ratelimit_impl.cc:117]: As I said it's not suitable for log spamming reasons. If there is no other stat, you can also potentially use `ENVOY_LOG_EVERY_POW_2` or similar.

Comment (zirain) [reply source/extensions/filters/common/ratelimit/ratelimit_impl.cc]: switch it to debug

**Comment by zirain** [comment]
> Thanks, small comment.
> 
> /wait

@mattklein123 please check the code again

**Review by mattklein123 — APPROVED** [review]
Thanks!

**Pull request merged**
